src/modules/evaluateExpression/evaluateExpression.ts

```diff
@@ -0,0 +1,128 @@
+interface IParameters {
+  [key: string]: any;
+  connection?: IConnection;
+}
+
+interface IQueryNode {
+  value?: string | number | boolean | object;
+  type?: string;
+  operator?: string;
+  children?: Array<IQueryNode>;
+}
+
+interface IConnection {
+  user?: string;
+  host?: string;
+  database?: string;
+  password?: string;
+  port?: number;
+  query?: any;
+}
+
+const defaultParameters: IParameters = {
+  connection: {},
+};
+
+export default async function evaluateExpression(
+  inputQuery: IQueryNode | string,
+  params = defaultParameters
+): Promise<string | number | boolean | any[]> {
+  // If input is JSON string, convert to Object
+  const query = typeof inputQuery === 'string' ? JSON.parse(inputQuery) : inputQuery;
+
+  // Base case
+  if (!query.children) {
+    return query.value;
+  }
+
+  // Recursive case
+  else {
+    const childrenResolved: any[] = await Promise.all(
+      query.children.map((child: IQueryNode) => evaluateExpression(child, params))
+    );
+
+    switch (query.operator) {
+      case 'AND':
+        return childrenResolved.reduce((acc: boolean, child: boolean) => {
+          return acc && child;
+        }, true);
+
+      case 'OR':
+        return childrenResolved.reduce((acc: boolean, child: boolean) => {
+          return acc || child;
+        }, false);
+
+      case 'CONCAT':
+        if (query.type === 'array') {
+          return childrenResolved.reduce((acc: any, child: any) => {
+            return acc.concat(child); // .flat(1) doesn't work for some reason
+          });
+        } else if (query.type === 'string' || !query.type) {
+          return childrenResolved.join('');
+        }
+        break;
+
+      case 'REGEX':
+        const str: string = childrenResolved[0];
+        const re: RegExp = new RegExp(childrenResolved[1]);
+        return re.test(str);
+
+      case '=':
+        return childrenResolved.every((child) => child === childrenResolved[0]);
+
+      case '!=':
+        return childrenResolved[0] !== childrenResolved[1];
+
+      case '+':
+        return childrenResolved.reduce((acc: number, child: number) => {
+          return acc + child;
+        }, 0);
+
+      case 'objectProperties':
+        try {
+          return params[childrenResolved[0].object][childrenResolved[0].property];
+        } catch {
+          return "Can't resolve object";
+        }
+
+      case 'pgSQL':
+        if (!params.connection) return 'No database connection provided';
+        return processPgSQL(childrenResolved, query.type, params.connection);
+
+      case 'graphQL':
+        if (!params.connection) return 'No database connection provided';
+        return processGraphQL(childrenResolved, params.connection);
+
+      // etc. for as many other operators as we want/need.
+    }
+  }
+  return 'No matching operators';
+}
+
+async function processPgSQL(queryArray: any[], queryType: string, connection: IConnection) {
```

Comment:
Move this function to Pg util function:
* processQuery

Reply:
This `evaluateExpression` module is going to be moved out into its own package eventually, so best not rely on unconnected modules, right? (Also, the PG function here is quite unique as the actual `query` is provided by the caller -- and it could be anything).

Reply:
Oh. I see your point. I was thinking if this would be part of what is shared between the Back-end core and the plugins. Not sure how to do this one either, so I think we should leave it for now and improve later on a PR dedicated of creating the basic structure to be accessible by the `evaluator` and `querySyntax` module.